prometheus_es_exporter/scheduler.py

```diff
@@ -1,21 +1,27 @@
 import time
 import logging
+import concurrent.futures
 
 log = logging.getLogger(__name__)
 
-
-def schedule_job(scheduler, interval, func, *args, **kwargs):
+def schedule_job(scheduler, executor, interval, func, *args, **kwargs):
     """
     Schedule a function to be run on a fixed interval.
 
     Works with schedulers from the stdlib sched module.
     """
 
     def scheduled_run(scheduled_time, *args, **kwargs):
-        try:
-            func(*args, **kwargs)
-        except Exception:
-            log.exception('Error while running scheduled job.')
+        def run_func(func, *args, **kwargs):
+            try:
+                func(*args, **kwargs)
+            except Exception:
+                log.exception('Error while running scheduled job.')
+
+        if executor != None:
+            executor.submit(lambda *args, **kwargs: run_func(func, *args, **kwargs), *args, **kwargs)
```

Comment:
I don't think the lambda is required?
```suggestion
        if executor is not None:
            executor.submit(run_func, func, *args, **kwargs)
```

Reply:
ah, good point!